extensions/wikia/SiteWideMessages/SpecialSiteWideMessages_body.php

```diff
@@ -46,8 +49,12 @@ class SiteWideMessages extends SpecialPage {
 
 	static $hasMessages = false;
 
+	/** @var UserGroupList */
+	private $userGroupList;
+
 	function  __construct() {
 		parent::__construct('SiteWideMessages' /*class*/, 'messagetool' /*restriction*/);
+		$this->userGroupList = Injector::getInjector()->get(UserGroupList::class);
 	}
 
 	function execute($subpage) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') The Cyclomatic Complexity of this function "execute" is 35 which is greater than 20 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1541)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') This function "execute" has 245 lines, which is greater than the 150 lines authorized. Split it into smaller functions. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS138)